File: mbfpp/code/ScalaExamples/src/main/scala/com/mblinn/mbfpp/functional/fb/PartialExamples.scala
package com.mblinn.mbfpp.functional.fb

object PartialExamples {

  def addTwoInts(intOne: Int, intTwo: Int) = intOne + intTwo

  val addFortyTwo = addTwoInts(42, _: Int)

  def taxForState(amount: Double, state: Symbol) = state match {
    // Simple tax logic, for example only!  
    case ('NY) => amount * 0.0645
    case ('PA) => amount * 0.045
    // Rest of states...
  }
  val nyTax = taxForState(_: Double, 'NY)
  val paTax = taxForState(_: Double, 'PA)
}

File: mbfpp/code/ClojureExamples/test/mbfpp/test/functional/fb/partial_examples.clj
(ns mbfpp.test.functional.fb.partial-examples
  (:use [mbfpp.functional.fb.partial-examples])
  (:use [midje.sweet]))

(fact "NY state tax is 0.0645"
      (ny-tax 100) => 6.45)

(fact "PA state tax is 0.045"
      (pa-tax 100) => 4.5)